pos_partner_firstname/static/src/xml/pos.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template" xml:space="preserve">
+
+    <t t-inherit="point_of_sale.ClientDetailsEdit" t-inherit-mode="extension">
+        <xpath expr="//input[@name='name']" position="attributes">
+            <attribute name="t-att-readonly">!props.partner.is_company</attribute>
+        </xpath>
+        <xpath expr="//div[@class='client-details-left']/div[1]" position="before">
```

Comment:
This field should be before the name field in my opinion, to have a similar display as in the back office.

![image](https://user-images.githubusercontent.com/3407482/161523202-9596d7cf-fd78-4668-bc30-bfbf944b803e.png)

Reply:
> This field should be before the name field in my opinion, to have a similar display as in the back office.
> 
> ![image](https://user-images.githubusercontent.com/3407482/161523202-9596d7cf-fd78-4668-bc30-bfbf944b803e.png)

I did this because the PoS typically you get private person

Reply:
my remark was about position : you have to choose before, the individual / company field, and then enter the name (or the last / first name). having the name field before is not intuitive.

> I did this because the PoS typically you get private person

I think it depends, in fact. There are B2B pos and B2C pos. my opinion is to have the same feature implemented in the different part of odoo.
Eventually, if you want to keep "individual" as default value, we should implement is as an option, in the pos.config, or better, load the default value of this field and use it in the PoS.